### PR TITLE
[CPU] Interpolate node migration on nGraph

### PR DESCRIPTION
**Pull request by mandrono**

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp [hunk 1]
```diff
@@ -1648,273 +1629,302 @@ SizeVector to5Dim(SizeVector casesDim) {
...
+                mode = InterpolateMode::linear_onnx;
+            } else {
+                mode = InterpolateMode::linear;
+            }
+        } else if (interpMode == ngraph::opset4::Interpolate::InterpolateMode::linear_onnx) {
+            mode = InterpolateMode::linear_onnx;
+        } else if (interpMode == ngraph::opset4::Interpolate::InterpolateMode::cubic) {
+            mode = InterpolateMode::cubic;
+        } else {
+            THROW_IE_EXCEPTION << errorPrefix << " does not support interpolate mode: " << ngraph::as_string(interpMode);
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp:1688]: This should be verified inside `isSupportedOperation`. The same for: interpCoordTransMode, interpNearestMode, interpShapeCalcMode.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp [hunk 2]
```diff
@@ -1648,273 +1629,302 @@ SizeVector to5Dim(SizeVector casesDim) {
...
-//        if (mayiuse(cpu::x64::avx2) && inputPrec == Precision::FP32) {
-//            pushDesc(MKLDNNMemory::GetPlainFormat(getParentEdgeAt(DATA_ID)->getDims()), jit_avx2);
-//        }
-//    }
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    setPostOps(attr, true);
+
+    Precision inputPrecision = getOriginalInputPrecisions()[DATA_ID];
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp:1839]: Please use `getOriginalInputPrecisionAtPort` instead

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp [hunk 3]
```diff
@@ -132,7 +119,15 @@ std::vector<CPUSpecificParams> filterCPUInfoForDevice() {
     }
     return resCPUParams;
 }
-/* ========== */
+
+std::vector<fusingSpecificParams> getFusingPatterns() {
+    std::vector<fusingSpecificParams> fusingPatterns;
+    fusingPatterns.push_back(emptyFusingSpec);
+    if (with_cpu_x86_sse42()) {
+        fusingPatterns.push_back(fusingSubtractPerChannel);
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp:127]: 1) Just didn't get why we need this condition? We don't have HW w/o sse42 support in validation.
2) I think we need to extend list of post ops here.

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp [hunk 4]
```diff
@@ -255,8 +228,7 @@ INSTANTIATE_TEST_CASE_P(smoke_InterpolateNN_Layout_Test, InterpolateLayerCPUTest
                 ::testing::Values(std::vector<size_t>({1, 21, 50, 60})),
                 ::testing::Values(CommonTestUtils::DEVICE_CPU)),
             ::testing::ValuesIn(filterCPUInfoForDevice()),
-            ::testing::ValuesIn(interpolateFusingParamsSet),
-            ::testing::ValuesIn(bf16EnforceFlags)),
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp:259]: Why you deleted `bf16EnforceFlags` config? Seems like BF16 is not covered now.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.h [hunk 5]
```diff
@@ -44,6 +44,11 @@ enum class InterpolateNearestMode {
     simple
 };
 
+enum class ShapeCalcMode {
```
[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.h]: changed

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp (hunk 1 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp]: moved into ```isSupportedOperation```

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp (hunk 2 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp]: fixed

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp (hunk 3 above)
Comment (mandrono) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp]: extended

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp (hunk 4 above)
Comment (mandrono) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/interpolate.cpp:259]: added ```BF16``` into validated precisions

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp [hunk 6]
```diff
@@ -1648,273 +1629,341 @@ SizeVector to5Dim(SizeVector casesDim) {
...
+        if (!interp) {
+            errorMessage = "Only opset4 Interpolate operation is supported";
+            return false;
+        }
+        const auto &interpAttr = interp->get_attrs();
+        const auto &interpMode = interpAttr.mode;
+        if (interpMode != ngraph::opset4::Interpolate::InterpolateMode::nearest &&
+            interpMode != ngraph::opset4::Interpolate::InterpolateMode::linear &&
+            interpMode != ngraph::opset4::Interpolate::InterpolateMode::linear_onnx &&
+                interpMode != ngraph::opset4::Interpolate::InterpolateMode::cubic) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp:1644]: Please use `one_of` here and in other places, It also would be worthwhile to add  using for `ngraph::opset4::Interpolate::InterpolateMode`

Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp]: done

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**